test/server/cards/02_SHD/leaders/HanSoloWorthTheRisk.spec.ts

```diff
@@ -0,0 +1,76 @@
+describe('Han Solo, Worth the Risk', function () {
+    integration(function (contextRef) {
+        describe('Han Solo\'s leader undeployed ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['cantina-braggart', 'twin-pod-cloud-car'],
+                        leader: 'han-solo#worth-the-risk',
+                        resources: 4,
+                    },
+                });
+            });
+
+            it('should play a unit from our hand, it costs 1 resource less and take 2 damage', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.hanSolo);
+                expect(context.player1).toBeAbleToSelectExactly([context.cantinaBraggart, context.twinPodCloudCar]);
+                context.player1.clickCard(context.cantinaBraggart);
+
+                expect(context.player1.exhaustedResourceCount).toBe(0);
+                expect(context.cantinaBraggart).toBeInZone('groundArena');
+                expect(context.cantinaBraggart.damage).toBe(2);
+                expect(context.hanSolo.exhausted).toBeTrue();
+
+                context.hanSolo.exhausted = false;
+                context.player2.passAction();
+
+                context.player1.clickCard(context.hanSolo);
+                // play automatically twin pod cloud car
```

Comment:
This is a good ability to confirm some window stuff. Could we set it up so that there is both a when played and when defeated ability that is triggerd by this? Looks like there's not a 1 or 2 hp unit that has both when played and when defeated, so maybe we need to use like Yularen or Gideon Hask or something.